libraries/heft-config-file/src/ConfigurationFileLoader.ts

```diff
@@ -0,0 +1,429 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as nodeJsPath from 'path';
+import { JSONPath } from 'jsonpath-plus';
+import { JsonSchema, JsonFile, PackageJsonLookup, Import, FileSystem } from '@rushstack/node-core-library';
+
+interface IConfigurationJson {
+  extends?: string;
+}
+
+/**
+ * @beta
+ */
+export enum InheritanceType {
+  /**
+   * Append additional elements after elements from the parent file's property
+   */
+  append = 'append',
+
+  /**
+   * Discard elements from the parent file's property
+   */
+  replace = 'replace'
+}
+
+/**
+ * @beta
+ */
+export enum PathResolutionMethod {
+  /**
+   * Resolve a path relative to the configuration file
+   */
+  resolvePathRelativeToConfigurationFile,
+
+  /**
+   * Resolve a path relative to the root of the project containing the configuration file
+   */
+  resolvePathRelativeToProjectRoot,
```

Comment:
FYI Jest and API Extractor both use tokens to make these nuances explicit.

Consider [this setting](https://api-extractor.com/pages/commands/config_file/#mainentrypointfilepath) for example:

![Capture](https://user-images.githubusercontent.com/4673363/91382942-c0683700-e7df-11ea-8a7d-d00d49d65b28.png)

It has a couple advantages:
- It allows the user to choose between multiple ways of resolving.
- It makes the resolution explicit, so nobody needs to memorize how each field gets resolved.

Given the complexity of Heft's config files, this might be worth implementing. It should go in a separate PR, though.

Reply:
Agreed.